ssh/Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM centos:7
+
+# Install OpenSSH
+RUN yum install -y epel-release.noarch
+RUN yum upgrade -y
+RUN yum -y install openssh-server openssh-clients sudo
+RUN yum install -y python3-pip && \
+    pip3 install -U dumb-init
+RUN yum clean all
+
+# Add user
+RUN useradd -s /bin/bash -g root -G wheel -u 1000 sshtest 
+RUN echo sshtest:Testing | chpasswd
+
+# Server config
+RUN /usr/bin/ssh-keygen -A
+COPY sshd_config /etc/ssh/sshd_config
+COPY sshrucio.key.pub /tmp/sshtest/.ssh/sshrucio.key.pub
+
+# Make directory to host the public keys
+RUN mkdir /root/.ssh
+RUN chown -R sshtest:root /root/.ssh; chmod -R 700 /root/.ssh
+
```

Comment:
@bziemons you're right, i this will be taking into consideration only root.
However, when I do ssh root@127.0.0.1, it still does not work:

```
(base) rakshita@rakshita-HP-Laptop-15:~/containers$ ssh-copy-id root@127.0.0.1
/usr/bin/ssh-copy-id: INFO: attempting to log in with the new key(s), to filter out any that are already installed
/usr/bin/ssh-copy-id: INFO: 4 key(s) remain to be installed -- if you are prompted now it is to install the new keys
root@127.0.0.1: Permission denied (publickey).
(base) rakshita@rakshita-HP-Laptop-15:~/containers$ ssh root@127.0.0.1
root@127.0.0.1: Permission denied (publickey).
(base) rakshita@rakshita-HP-Laptop-15:~/containers$ ssh-copy-id root@127.0.0.1
/usr/bin/ssh-copy-id: INFO: attempting to log in with the new key(s), to filter out any that are already installed
/usr/bin/ssh-copy-id: INFO: 4 key(s) remain to be installed -- if you are prompted now it is to install the new keys
root@127.0.0.1: Permission denied (publickey).
```
This prompt refers to entering the password for the server right? but in spite of disabling the password in sshd_config, this error comes. Should I be restarting the server after updating the sshd_config in the 17th line?

Reply:
so i tried restarting, still gives the same error.

Reply:
`/root/.ssh` **must** be owned by `root:root` (along other restrictions in the name of security) for public key authentication

Reply:
@bziemons right, so if the extra user is removed and it is simply done as USER root, it should ideally work right?

Reply:
yes, that would be fine :) We can also have a quick chat about it. I do see a way how this could work.

Reply:
@bziemons its still failing as USER root.